crwatcher/watcher.go

```diff
@@ -114,20 +116,47 @@ func (cw *CRWatcher) setupHandler(con ResourceController) {
 	cw.handler = cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
 			r := obj.(*unstructured.Unstructured)
+			if cw.Config.Filter != "" && !cw.hasFilter(r) {
+				return
+			}
 			con.ResourceAdded(r)
 		},
 		DeleteFunc: func(obj interface{}) {
 			r := obj.(*unstructured.Unstructured)
+			if cw.Config.Filter != "" && !cw.hasFilter(r) {
+				return
+			}
 			con.ResourceDeleted(r)
 		},
 		UpdateFunc: func(oldObj, newObj interface{}) {
 			oldR := oldObj.(*unstructured.Unstructured)
 			newR := newObj.(*unstructured.Unstructured)
-			con.ResourceUpdated(oldR, newR)
+			if cw.Config.Filter != "" {
```

Comment:
can we remove all the logic here and have this only call our update and that triggers the external c/u/d

Reply:
Done